app/models/queen.rb

```diff
@@ -1,3 +1,15 @@
 class Queen < Piece
 
-end
+  def is_valid?(new_x, new_y)
+    return false if not super
+    return is_diagnoal?(new_x, new_y) || is_straight?(new_x, new_y)
+  end
+
+  def is_diagnoal?(new_x, new_y)

```

Comment:
what about naming it `is_valid_diagonal?` and `is_valid_straight?`? Coming from the valid? method it makes sense, but if you just see the method alone, the name can be a bit misleading..

Reply:
Yeah that totally makes sense. I'll add that change.